traffic_ops/experimental/go-api/tenant/EndPointSeeder.go

```diff
@@ -0,0 +1,202 @@
+
+    "github.com/gorilla/mux"
+  _ "github.com/lib/pq"
+)
+
+
+//TODO(nirs) polymorphisem
+
+/////BASE - COMMON TO ALL SEEDERS (needs to move to another package)
+/// Defintions
+type EndpointSeeder struct{
+    myDb        *sql.DB
+}
+
+/// Interface
+func NewEndpointSeeder(aDb *sql.DB) *EndpointSeeder {
+    endPointSeeder := new(EndpointSeeder)
+    endPointSeeder.myDb = aDb
+    return endPointSeeder
+}
+
+func (aEndpointSeeder *EndpointSeeder) Seed (aRouter *mux.Router, aApiPrefix string) {
+
+    aRouter.HandleFunc(aApiPrefix+"/tenants", aEndpointSeeder.getListEndpoint).Methods("GET")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.getEndpoint).Methods("GET")
+    aRouter.HandleFunc(aApiPrefix+"/tenants", aEndpointSeeder.createEndpoint).Methods("POST")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.updateEndpoint).Methods("PUT")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.deleteEndpoint).Methods("DELETE")
+}
+
+///Common utilities
+func (aEndpointSeeder *EndpointSeeder) checkErr(aError error) {
+    //TODO log
+    if aError != nil {
+        panic(aError)
+    }
+}
+
+
+/////Class specific
+
+type Tenant struct {
+    Id        int       `json:"id"`
+    Name      string    `json:"name"`
+    Active    bool      `json:"active"`
+    ParentId  int       `json:"parentId"`
+}
+
+func (aEndpointSeeder *EndpointSeeder) getEndpoint(aResponseWriter http.ResponseWriter, aRequest *http.Request) {
+    params := mux.Vars(aRequest)
+
+    var id int
+    var name string
+    var active bool
+    var parent_id sql.NullInt64
+    var updated time.Time;
+    
+    err := aEndpointSeeder.myDb.QueryRow("SELECT * FROM tenant WHERE id="+params["id"]).Scan(&id, &name, &active, &parent_id, &updated)
+    if err == sql.ErrNoRows{//TODO make it work
+        aResponseWriter.WriteHeader(http.StatusNotFound)
+        fmt.Fprint(aResponseWriter, params["id"]+" not found")
+        return
+    }
+    aEndpointSeeder.checkErr(err)
+    
+    var parent_id1 int
+    if parent_id.Valid {
+        parent_id1 = int(parent_id.Int64)
+    } else {
+       parent_id1 = 0        
+    }
+
+    json.NewEncoder(aResponseWriter).Encode(Tenant{Id: id, Name: name, Active: active, ParentId: parent_id1})
+    return
+}
+    
+
+ 
+func (aEndpointSeeder *EndpointSeeder) getListEndpoint(aResponseWriter http.ResponseWriter, aRequest *http.Request) {
+    var tenants []Tenant
+    rows, err := aEndpointSeeder.myDb.Query("SELECT * FROM tenant")
+    aEndpointSeeder.checkErr(err)
+    
+    for rows.Next() {
+        var id int
+        var name string
+        var active bool
+        var parent_id sql.NullInt64
+        var updated time.Time;
+            
+        err = rows.Scan(&id, &name, &active, &parent_id, &updated)
+        aEndpointSeeder.checkErr(err)
+
+        var parent_id1 int
+        if parent_id.Valid {
+            parent_id1 = int(parent_id.Int64)
+        } else {
+            parent_id1 = 0        
```

Comment:
This is unnecessary, Go defaults uninitialized integral values to 0. 

Actually, the if statement isn't necessary at all, you can simply `parent_id1 := int(parent_id.Int64)`, because `parent_id.Int64` will be default-initialized to 0, even if it isn't valid.